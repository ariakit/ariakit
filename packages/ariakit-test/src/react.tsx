import type { ReactElement } from "react";
import { StrictMode } from "react";
import * as ReactTestingLibrary from "@testing-library/react";
import { flushMicrotasks, nextFrame, wrapAsync } from "./__utils.js";

export * from "./index.js";

export interface RenderOptions
  extends Omit<ReactTestingLibrary.RenderOptions, "queries"> {
  strictMode?: boolean;
}

export async function render(ui: ReactElement, options?: RenderOptions) {
  const wrapper = (props: { children: ReactElement }) => {
    const Wrapper = options?.wrapper;
    const element = Wrapper ? <Wrapper {...props} /> : props.children;
    if (!options?.strictMode) return element;
    return <StrictMode>{element}</StrictMode>;
  };

  return wrapAsync(async () => {
    const { unmount } = ReactTestingLibrary.render(ui, { ...options, wrapper });
    await flushMicrotasks();
    await nextFrame();
    await flushMicrotasks();
    return unmount;
  });
}
