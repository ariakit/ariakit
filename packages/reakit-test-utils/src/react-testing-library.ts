export {
  act,
  fireEvent,
  render,
  wait,
  waitForDomChange,
  waitForElement,
  waitForElementToBeRemoved
} from "@testing-library/react";
