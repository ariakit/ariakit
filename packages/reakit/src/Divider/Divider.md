It serves the same function as HTML's `<hr>`, to separate content. Use prop `vertical` or `horizontal` to toggle Divider's orientation. It defaults to a 1px wide horizontal line.

```jsx
import { Divider } from "reakit";

<Divider />
```

Basic styling via props, with `vertical`.

```jsx
import { Flex, Block, Divider } from "reakit";

<Flex
  backgroundColor="rgb(85.9%, 54.3%, 30.3%)"
  color="white"
  padding={8}
  justifyContent="center"
>
  <Block>Apples</Block>
  <Divider vertical />
  <Block>Oranges</Block>
</Flex>
```
