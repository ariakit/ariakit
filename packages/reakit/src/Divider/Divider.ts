import * as PropTypes from "prop-types";
import { ifProp, theme } from "styled-tools";
import styled, { css } from "../styled";
import Box, { BoxProps } from "../Box";

export interface DividerProps extends BoxProps {
  vertical?: boolean;
}

const Divider = styled(Box)<DividerProps>`
  border-color: currentColor;
  border-style: solid;
  opacity: 0.2;

  ${ifProp(
    "vertical",
    css`
      margin: 0 1em;
      min-height: 100%;
      width: 0;
      border-width: 0 0 0 1px;
    `,
    css`
      margin: 1em 0;
      height: 0;
      border-width: 1px 0 0 0;
    `
  )};

  ${theme("Divider")};
`;

// @ts-ignore
Divider.propTypes = {
  vertical: PropTypes.bool
};

Divider.defaultProps = {
  use: "hr"
};

export default Divider;
