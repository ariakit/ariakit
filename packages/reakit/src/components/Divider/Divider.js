import PropTypes from "prop-types";
import styled, { css } from "styled-components";
import { ifProp, prop } from "styled-tools";
import as from "../../enhancers/as";
import Base from "../Base";

const Divider = styled(Base)`
  border: 1px solid currentcolor;
  opacity: 0.2;

  ${ifProp(
    "vertical",
    css`
      margin: 0 1rem;
      min-height: 100%;
      width: 0;
      border-width: 0 0 0 1px;
    `,
    css`
      margin: 1rem 0;
      height: 0;
      border-width: 1px 0 0 0;
    `
  )};

  ${prop("theme.Divider")};
`;

Divider.propTypes = {
  vertical: PropTypes.bool
};

export default as("div")(Divider);
