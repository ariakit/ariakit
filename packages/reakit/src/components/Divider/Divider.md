It serves the same function as HTML's `<hr>`, to separate content. Use prop `vertical` or `horizontal` to toggle Divider's orientation. It defaults to a 1px wide horizontal line.

```jsx
<Divider />
```

Basic styling via props, with `vertical`.

```jsx
import { Flex, Block } from "reakit";

<Flex
  backgroundColor="rgb(85.9%, 54.3%, 30.3%)"
  color="white"
  padding={8}
  justifyContent="center"
>
  <Block>Apples</Block>
  <Divider vertical />
  <Block>Oranges</Block>
  <Divider vertical />
  <Block>Grapes</Block>
  <Divider vertical />
  <Block>Strawberries</Block>
</Flex>
```
