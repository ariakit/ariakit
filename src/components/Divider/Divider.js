import React from 'react'
import styled from 'styled-components'
import PropTypes from 'prop-types'
import as from '../../enhancers/as'
import { compose, setDisplayName, setPropTypes, defaultProps } from 'recompose'

const Divider = styled.div.attrs({
  className: props=>props.horizontal && 'horizontal' || props.vertical && 'vertical'
})`
  
  &.horizontal {
    margin: 1rem 0;
    height: 0;
    border-top: 1px solid rgba(34, 36, 38, 0.15);
    border-bottom: 1px solid rgba(255, 255, 255, 0.1);
  }
  &.vertical {
    margin: 0 1rem;
    height: 100%;
    width: 0;
    display: inline-block;
    border-left: 1px solid rgba(34, 36, 38, 0.3);
    border-right: 1px solid rgba(255, 255, 255, 0.1);
  }
`

const enhance = compose(
  setDisplayName('Divider'),
)

export default enhance(Divider)
