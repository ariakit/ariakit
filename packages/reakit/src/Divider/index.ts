import Divider from "./Divider";

export * from "./Divider";

export default Divider;
