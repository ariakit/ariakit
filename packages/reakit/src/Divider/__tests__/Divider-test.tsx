import * as React from "react";
import { render } from "react-testing-library";
import Divider from "../Divider";

test("html attrs", () => {
  const { getByTestId } = render(
    <Divider id="test" aria-label="test" data-testid="test" />
  );

  expect(getByTestId("test")).toHaveAttribute("id", "test");
  expect(getByTestId("test")).toHaveAttribute("aria-label", "test");
});

test("styled", () => {
  const { container } = render(<Divider />);
  expect(container.firstChild).toMatchSnapshot();
});

test("styled vertical", () => {
  const { container } = render(<Divider vertical />);
  expect(container.firstChild).toMatchSnapshot();
});
