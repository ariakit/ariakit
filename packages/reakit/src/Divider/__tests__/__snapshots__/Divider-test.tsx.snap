// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`styled 1`] = `
.c1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: unset;
  color: inherit;
}

.c0 {
  border-color: currentColor;
  border-style: solid;
  opacity: 0.2;
  margin: 1em 0;
  height: 0;
  border-width: 1px 0 0 0;
}

<hr
  class="c0 c1"
/>
`;

exports[`styled vertical 1`] = `
.c1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: unset;
  color: inherit;
}

.c0 {
  border-color: currentColor;
  border-style: solid;
  opacity: 0.2;
  margin: 0 1em;
  min-height: 100%;
  width: 0;
  border-width: 0 0 0 1px;
}

<hr
  class="c0 c1"
/>
`;
